cylc/flow/simulation.py

```diff
@@ -29,12 +34,61 @@
 )
 from cylc.flow.wallclock import get_current_time_string
 
-from metomi.isodatetime.parsers import DurationParser
+from metomi.isodatetime.parsers import DurationParser, TimePointParser
 
 if TYPE_CHECKING:
     from queue import Queue
+    from cylc.flow.broadcast_mgr import BroadcastMgr
     from cylc.flow.cycling import PointBase
     from cylc.flow.task_proxy import TaskProxy
+    from cylc.flow.workflow_db_mgr import WorkflowDatabaseManager
+
+
+@dataclass
+class ModeSettings:
+    """A store of state for simulation modes.
+
+    Used instead of modifying the runtime config.
+    """
+    simulated_run_length: float = 0.0
+    sim_task_fails: bool = False
+
+    def __init__(
+        self,
+        itask: 'TaskProxy',
+        broadcast_mgr: 'BroadcastMgr',
+        db_mgr: 'Optional[WorkflowDatabaseManager]' = None
+    ):
+        overrides = broadcast_mgr.get_broadcast(itask.tokens)
+        if overrides:
+            rtconfig = pdeepcopy(itask.tdef.rtconfig)
+            poverride(rtconfig, overrides, prepend=True)
+        else:
+            rtconfig = itask.tdef.rtconfig
+        self.simulated_run_length = (
+            get_simulated_run_len(rtconfig))
+        self.sim_task_fails = sim_task_failed(
+            rtconfig['simulation'],
+            itask.point,
+            itask.submit_num
+        )
+
+        # itask.summary['started_time'] and mode_settings.timeout need
+        # repopulating from the DB on workflow restart:
+        started_time = itask.summary['started_time']
+        if started_time is None and db_mgr:
+            started_time = int(
+                TimePointParser()
```

Comment:
Initialising the `TimePointParser` for every task is going to get expensive.

We carefully cache the instance elsewhere in the code. Maybe try using the `WorkflowSpecifics` cache?

Reply:
Turned out to be a bit trickier to fix this than expected, because I wasn't putting the time into the DB in a format which the `get_unix_time_from_time_string` was happy with.